src/orion/serving/benchmarks_resource.py

```diff
@@ -0,0 +1,92 @@
+"""
+Module responsible for the benchmarks/ REST endpoint
+=====================================================
+
+Serves all the requests made to benchmarks/ REST endpoint.
+
+"""
+import json
+from typing import Optional
+
+from falcon import Request, Response
+
+from orion.core.worker.experiment import Experiment
+from orion.core.worker.trial import Trial
+from orion.serving.parameters import retrieve_benchmark, verify_query_parameters
+from orion.serving.responses import build_benchmark_response, build_benchmarks_response
+
+
+class BenchmarksResource:
+    """Handle requests for the benchmarks/ REST endpoint"""
+
+    def __init__(self, storage):
+        self.storage = storage
+
+    def on_get(self, req: Request, resp: Response):
+        """Handle the GET requests for benchmarks/"""
+        benchmarks = self.storage.fetch_benchmark({})
+
+        response = build_benchmarks_response(benchmarks)
+        resp.body = json.dumps(response)
+
+    def on_get_benchmark(self, req: Request, resp: Response, name: str):
+        """
+        Handle GET requests for benchmarks/:name where `name` is
+        the user-defined name of the benchmark
+        """
+        verify_query_parameters(req.params, ["assessment", "task", "algorithms"])
+        assessment = req.get_param("assessment")
+        task = req.get_param("task")
+        algorithms = req.get_param_as_list("algorithms")
+        benchmark = retrieve_benchmark(
+            self.storage,
+            name,
+            assessment=assessment,
+            task=task,
+            algorithms=algorithms,
+        )
+
+        response = build_benchmark_response(benchmark, assessment, task, algorithms)
+        resp.body = json.dumps(response)
+
+
+def _find_latest_versions(experiments):
```

Comment:
Looks like the functions below where copy pasted from experiments_resource but are not used. Sorry, that was me. Could you please remove them? :grimacing:

Reply:
Done !

Reply:
Thanks! I think the other functions are not used as well.